extension/src/providers/gotoProvider.ts

```diff
@@ -0,0 +1,67 @@
+import * as vscode from 'vscode';
+import { AutoLispExt } from '../extension';
+import { Sexpression } from '../format/sexpression';
+import { ReadonlyDocument } from '../project/readOnlyDocument';
+
+interface LocatedItem {
+	line: number;
+	column: number;
+	fspath: string;
+}
+
+export class AutolispDefinitionProvider implements vscode.DefinitionProvider{
+	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Location | vscode.Location[]> {
+		const editor: vscode.TextEditor = vscode.window.activeTextEditor;		
+		let selected: string =  editor.document.getText(editor.selection);
+		if (selected === "") {
+			ReadonlyDocument.getMemoryDocument(document).atomsForest.forEach(sexp => {
+				if (sexp instanceof Sexpression && sexp.contains(position)){
+					const atom = sexp.getAtomFromPos(position);
+					if (atom) {
+						selected = atom.symbol;
+					}		
+				}
+			});			
+		}
+		try {
+			return this.findGoToLocations(selected);
+		} catch (error) {
+			return;	// I don't believe this requires a localized error since VSCode has a default "no definition found" response
+		}
+	} 
+
+	// This was designed around the idea of "preference" for opened and workspace documents
+	private findGoToLocations(searchFor: string): vscode.Location[] {
+		const locations: vscode.Location[] = [];
+		let possible = this.findDefunMatches(searchFor, AutoLispExt.Documents.OpenedDocuments.concat(AutoLispExt.Documents.ProjectDocuments));
+		if (possible.length === 0) {
+			possible = this.findDefunMatches(searchFor, AutoLispExt.Documents.WorkspaceDocuments);
+		}
+		possible.forEach(item => {			
```

Comment:
If the opened files are not in current workspace, and are not included in the project file, it will not find in those files?

Reply:
I am doing a concat on the `AutoLispExt.Documents.OpenedDocuments` and `AutoLispExt.Documents.ProjectDocuments` lists, so I believe these will find non-workspace documents as long as they have been "activated" in the interface. If the tab was left open from a previous sessions, was not in the workspace and has not been activated in the new session, then the `AutoLispExt.Documents.OpenedDocuments` variable would not know about the document. Agreed, there are scenarios where it would not be useable, but I don't think we can do anything about them until the VSCode team provide access to what is actually opened in the UI.